src/topology/mod.rs

```diff
@@ -593,6 +597,11 @@ impl RunningTopology {
 
     fn remove_outputs(&mut self, name: &str) {
         self.outputs.remove(name);
+
+        #[cfg(feature = "api")]
+        self.tap_sinks.iter().for_each(|tap_sink| {
```

Comment:
If outputs are removed, we preemptively fire off a `component_not_matched` to all underlying taps. We don't need to check whether a given tap holds the component or not; that's done in the sink itself.

Taps that are actively watching for a given component being available are then sent a log event notification to say that the component no longer matches.